CONTRIBUTING.md

```diff
@@ -49,8 +49,26 @@ go test -v ./pkg/...
 
 * Branch from the master branch and, if needed, rebase to the current master branch before submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase your changes.
 
-* Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
-
 * If your patch is not getting reviewed or you need a specific person to review it, you can @-reply a reviewer asking for a review in the pull request or a comment.
 
 * Add tests relevant to the fixed bug or new feature.
+
+### New features
+Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
+
+### Bug fixes
+Please make all changes in one commit if possible. Include `closes #12345` in bottom of the commit message.
```

Comment:
> Include `closes #12345`

This will work when there are a single commit and no changes needed. But how about when we require changes - rebase?

PR's should definitely have `fixes #<issue nr>`  or `closes #<issue nr>` in description for both new features and bug fixes.

Reply:
Bugfixes that need more work should IMO be rebased into one commit before we merge it. 

One of the downsides of cherry-picking is that it makes it harder to see what releases a patch is included in since the cherry-pick will have a new git commit sha. But if the commits contain a link to the issue it will be possible to find commits in master and release branches.

When the intention is only to target the master branch, having fixes/closes in the pr description is the preferred way. (I'll include that in this pr)

Reply:
> One of the downsides of cherry-picking is that it makes it harder to see what releases a patch is included in since the cherry-pick will have a new git commit sha. But if the commits contain a link to the issue it will be possible to find commits in master and release branches.

This to me sounds like we maybe should consider squashing all commits when merging. Because if you have `Fixes #12345` in PR description it will per default be included as commit message when "squash and merge" through Github. Cherry picking PR's would then be simple to automate given it's easy to retrieve the merged commit from PR.

Reply:
Lets move this discussion to the future...

Reply:
Yea we dont have to solve that right now